### DISCUSSION TITLE
List of potential default fontawesome icons

### DISCUSSION THREAD

**Discussion by blakadder** [HASwitchPlate/openHASP#41] [Ideas]
List of fontawesome icons that (sh/c)ould be default in binaries

door-closed
door-open
home
microphone
angle-???
arrow-???
backward
ban
battery-???
bath
bars
bell
bed
bolt
car
caret-???
check-???
chart-???
chevron-???
cloud-???
coffee
cog
cogs
desktop
folder-???
forward
info
info-circle
key
lightbulb
minus-???
moon
music
pause-???
play-???
plug
plus-???
poo
portrait
puzzle-piece
question-???
redo
reply
shapes
signal
skull
sliders-h
snowflake
square
star
step-???
stop
stopwatch
sun
temperature-???
thermometer-???
toggle-???
toilet
trash
tv
umbrella
undo
unlock
user
utensils
volume-???
warehouse
wifi

**Comment by fvanroie** [comment]
OK, I'll have a look at the filesize this adds to the firmware .bin
At the same time, we can ommit some icons that don't make sense... to save some space there.

Do we keep Montserrat as the default font on ESP32? Because if we go custom for the built-in fonts, it's the same effort to swap out the entire default font for another.

**Comment by blakadder** [comment]
Personally I don't really like Montserrat. I'm partial towards Rubik, Roboto and Roboto Condensed. I like using narrower fonts because of limited real estate

**Comment by fvanroie** [comment]
OK, I will try all three and the one with the smallest memory size will probably win the battle...

**Comment by blakadder** [comment]
What is the default font lvgl comes with?

**Comment by fvanroie** [comment]
The current *built-in* fonts in hasp-lvgl are the default fonts in lvgl: Montserrat in various size and 2 dozen of extra icons.
I haven't customized any of those fonts yet, but they provide an option to compile your own so that's what I'm trying now.

**Comment by blakadder** [comment]
Keep the default Montserrat then

**Comment by fvanroie** [comment]
By adding new icons the default font also needs to be re-rendered anyway, so we can swap it at the same time. So far, it seems Roboto Condensed and Roboto use the least amount of storage/memory.

I have made a temporary list to give us a better overview of what the iconset would look like: https://haswitchplate.github.io/openHASP-docs/#fonts

That definitely looks more like a set that is geared towards iot applications.

Here's the look of Roboto Condensed in 12 16 22 and 28pt:
![image](https://user-images.githubusercontent.com/15969459/104532725-33e4d680-5611-11eb-8d97-da4f939c502b.png)

**Comment by fvanroie** [comment]
This list will be superseeded in an upcoming release by a set of Material Design Icons: https://haswitchplate.github.io/openHASP-docs/0.5/#fonts-mdi

**Reply by RossiRPI** [reply]
Are there plans to get larger icon sizes for the ESP32 release? Size 28 looks quite small on a 320x480 screen.

**Reply by fvanroie** [reply]
We're experimenting with this, but nothing concrete yet. You can have a look at the mdi-icons branch which includes font sizes up to 40. You will need to change the lv_conf_v7.h and compile yourself to use these.

**Reply by schabau** [reply]
There's a lot of solar energy, radiators in all variations, arrows in every shape and direction, and every conceivable lamp. But not a single open window is to be found. If the selection depends on only a few people, any change request is difficult to implement. The font with all its symbols is present in the code, but it seems the unreleased symbols cannot be accessed. (materialdesignicons-webfont.ttf).

**Reply by fvanroie** [reply]
This is a very old 0.5.x thread before TTF fonts were supported.... In the mean time at [least 4 different window icons](https://openhasp.com/0.7.0/design/fonts/#icons) are available in 0.7.0 by default. More can be added by uploading your own (custom) fonts.

**Reply by schabau** [reply]
> This is a very old 0.5.x thread before TTF fonts were supported....

Hello,

The situation with the integrated icons has only improved marginally (thankfully, weather icons are now standard in the firmware), and the issue is still open and relevant. Especially since modern displays have significantly more memory available, which should be utilized. So why create a new thread for an old problem?
Even today (0.7.0), I only see one closed window icon, "window-closed-variant": "0xF11DB=>0xF1DB," among the standard icons.
Everything else isn't a window, but rather a shutter or blind. In home automation, there's a clear difference between open and closed shutters and windows. Of course, a trade-off must always be made between available memory and available icons, but loading additional fonts onto the display for just a few less icons isn't a better solution in my opinion.

After many attempts and multiple reinstalls, I finally managed to compile my firmware myself on a Debian VM, and with 8kB more LVGL, my Guition JC8048W50N is now working very reliably. Previously, my display kept crashing with out-of-memory errors, even though I only use a few colors and no additional fonts. Apparently, the number of pages was too high, and that's with 5 MiB of unused PSRAM, which I still lack the knowledge and information on how to use effectively.

Since the `materialdesignicons-webfont.ttf` font is already present in the code, shouldn't it be possible to add more icons to the firmware?
What would I need to do to add, for example, the icon "window-open-variant": "0xF11DC=>0xF1DC" to my firmware? I can't see how to do this in `include/user_config_override.h`. It seems to be about adding fonts, but in this example, I only want the one icon from the existing font to be available on the display.

Best regards,
Detlef Paschke

**Reply by fvanroie** [reply]
> but loading additional fonts onto the display for just a few less icons isn't a better solution in my opinion.

We can agree to disagree. There are thousands of icons and dozens of icon fonts. openHASP doesn't guarantee that your specific use-case is covered by the built-in icons. The list is provided "as is". Any list we decide will miss some icons a user will "really need" and is "still missing". Icons are not added per single user request, since other options exist. If we add an icon, another needs to go which causes other issues. The problem isn't PSRAM, it's program size of the firmware file...

You can easily upload a TTF font with your icons, as per documentation. Use a tool like FontForge to trim the number of glyphs/icons/filesize down to the ones you need. That's how I created a minimal "openhasp.tff" font that is included in the firmware. No other source TTF is used, unless you upload it to the filesystem.

You can modify the default `openhasp.tff` font to include you own personal favorite icons too, which is then compiled into your customized firmware. That's what OSS projects are for.

If you only need one extra icon, you could consider adding it as a `.png` image instead.
